client/src/util/fetchFromAPI.js

```diff
@@ -0,0 +1,18 @@
+export const queryAPI = async (options) => {
+  const { url, onSuccess, onError, method } = options;
+  const result = {
+    data: null,
+    error: null,
+  };
+  const response = fetch(url, method);
+  if (response.ok) {
+    result.data = await response.json();
+    result.data = response.json({ message: "Payload is valid" });
+    onSuccess();
```

Comment:
How do you feel about passing the `result` object into this `onSuccess` method?

```suggestion
    onSuccess(result);
```